spec/Spec/BlsHelpers.cry

```diff
@@ -62,6 +62,34 @@ g1_multi_exp : {len} (fin len, len >= 1) => [len]G1Point -> [len]UInt64 -> G1Poi
 g1_multi_exp points scalars = results!0 where
     results = [G1_INFINITY] # [g1_add (results@i) (g1_multi (points@i) (scalars@i)) | i <- [0 .. len-1]]
 
+/**
+ * Helper function for decompressing a G1 or G2 compresed point.
+ *
+ * The most-significant three bits of a G1 or G2 encoding should be masked away before
+ * the coordinate(s) are interpreted.
+ * These bits are used to unambiguously represent the underlying element.
+ * The format: (cFlag, bFlag, aFlag, x)
+ *  - cFlag: is '1' to indicated a compressed point, '0' otherwise.
+ *  - bFlag: is '1' to indicate a point at infinity, '0' otherwise.
+ *     - NOTE: the cFlag must also be set to indicate an intentionally
+ *             compressed infinity point.
+ *  - aFlag: is used to store the most significant bit of the y-coordinate.
+ *  - x: is the x-coordinate of the point.
+ *
+ * @see https://github.com/ethereum/py_ecc/blob/main/py_ecc/bls/point_compression.py#L43
+ *
+ * ```repl
+ * :prove get_flags`{8} 0xC0 == (True, True, False)
+ * :prove get_flags`{384} (0xE0 # (zero:[384-8])) == (True, True, True)
+ * ```
+ */
+get_flags : {n} (fin n, n >= 3) => [n] -> (Bit, Bit, Bit)
```

Comment:
This function gets used for the bytes representation of both G1 points (384 bit value) and G2 bytes (double that size).

BUT, I could split the G2 bytes into two 384-bit values before calling `get_flags` instead of calling `get_flags` on the full value and then on the lower 384 bits. What do you think?

Reply:
I'm not sure I understand -- do you always need to call `get_flags` twice for G2?

If so I would say yes, hardcode 384. If not then it's okay to leave it generic but you might consider explaining in the docs why it's not identical to the spec (or e.g. note if eventually you're going to add another link to a G2 version with a different hardcoded value).

Reply:
Yes, always twice because the G2 compression bytes are just two 384 bit values. I need to check the Python code again. I'm actually just now needing to play with G2 points, so I'm going to file and issue for this. Thanks!